core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -107,4 +122,16 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
 
     return newFiles;
   }
+
+  public static Integer existingDataFiles(Snapshot snapshot) {
```

Comment:
I don't think this should return an `Integer`. There isn't a case that I can think of where the value should be null.